### ISSUE TITLE
run the benchmark nightly on Hetzner runners

### ISSUE THREAD

**Issue by problame** [neondatabase/neon#6476]
We currently run small configurations of the `test_pageserver_max_throughput_getpage_at_latest_lsn` benchmark on every merge from `main`, as part of the `benchmarks` job in `build_and_test.yml`.

We can't run large configurations because it consumes too much CI time  after each merge into `main` (https://github.com/neondatabase/neon/pull/6477)

More context: https://neondb.slack.com/archives/C06750CD9U5/p1706200941278869

# DoD

The benchmark doesn't run on each merge from `main`, but only once per night.

It runs on an otherwise idle Hetzner runner.

Results are recorded the same way as the other nightly benchmarks from `benchmarking.yml` so we can track regressions in Grafana.

Results carry as annotation the Hetzner runner type, so we can distinguish results from Hetzner runners from the results from EC2 runners to which we'll move in #6297

# Future Work

This is a preliminary to https://github.com/neondatabase/neon/issues/6297 where we want to run the benchmark on the more practically relevant and more predictable EC2  instances.

**Comment by problame** [comment]
Obsolete, we now run pagebench nightly on ec2 https://github.com/neondatabase/neon/issues/6297
